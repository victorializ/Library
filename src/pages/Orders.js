import React from 'react';
import { connect } from 'react-redux';
import { Header } from 'semantic-ui-react';

import { constants  } from '../i18n';
import { CurrentUser, LoadingComponent, OrdersList } from '../components';
import { useRequest } from '../hooks';
import { getAllOrders } from '../services/http-client';

function Orders({userId}) {  
  const response = useRequest(getAllOrders, [userId]);
  return (
    <>
      <CurrentUser />   
      <Header>{constants.orders}</Header>
      <div className='orders__list'>
        <LoadingComponent {...response}>
          <OrdersList data={response.data} />
        </LoadingComponent>
      </div>
    </>
  );
}

const UserOrders = connect(({user}) => user)(Orders);

export { Orders, UserOrders };