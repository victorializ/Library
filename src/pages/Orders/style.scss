@import '../../assets/variables.scss';

.orders {
    &__list {
        @include column-flex-container;
        height: 100%;
    }
}