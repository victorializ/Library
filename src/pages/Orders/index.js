import { connect } from 'react-redux';

import { Orders } from './Orders';

const UserOrders = connect(({user}) => user)(Orders);

export { UserOrders };