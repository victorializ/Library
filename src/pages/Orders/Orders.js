import React, { Fragment } from 'react';

import { Card, Header } from 'semantic-ui-react'

import { constants  } from '../../i18n';
import { colors } from '../../assets/semantic-colors'
import { CurrentUserIcon, useRequest, WithRequest } from '../../components';
import moment from 'moment';

import { formatData } from '../../services/format-data/order';
import { getAll } from '../../services/http-client/orders';

import './style.scss';

function Orders({id: userId}) {
    
  const [ orders, error ]  = useRequest(getAll, userId, userId);
  return (
      <Fragment>
        <CurrentUserIcon />   
        <div className='orders__list'>
            <WithRequest 
                error={error}
                data={orders} 
                WrappedComponent={OrdersList} 
            /> 
        </div>
      </Fragment>
  )
}

function OrdersList({data}) {
    return (
        <Fragment>
            <Header>{constants.orders}</Header>
            <Card.Group>
                {
                data.map(element => 
                    <Order key={element.bookingId} {...formatData(element)} />) 
                }
            </Card.Group>
        </Fragment>
    )
}

function Order({
    name, 
    bookYear,
    date, 
    isFinished
}) {
    const statuses = {
        finished: {
            message: constants.returned, 
            color: colors.grey
        }, 
        active: {
            message: `${constants.return} ${moment(date).calendar()}`, 
            color: colors.primary
        }, 
        expired: {
            message: constants.expired, 
            color: colors.red
        }
    };
    const isReturnDateAfterToday = () => {
        const now = moment();
        const returnDate = moment(date);
        return returnDate.diff(now) > 0;
    }
    const status = () => 
        isFinished ? 
            statuses.finished 
                :
            isReturnDateAfterToday() ?
                statuses.active
                    :
                statuses.expired;
   
    return (
        <Card className='book' color={status().color}>
            <Card.Content>
                <Card.Header>
                    {name}
                </Card.Header>
                <Card.Meta>
                    {bookYear}
                </Card.Meta>
                <Card.Description>
                    {`${constants.status}: ${status().message}`}
                </Card.Description>
            </Card.Content>
        </Card>
    )
}

export { Orders };