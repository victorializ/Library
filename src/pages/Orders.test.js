import React from 'react';
import { shallow } from 'enzyme';

import { Orders } from './Orders';

describe('Orders page component', () => {
    it('renders correctly', () => {
        expect(shallow(<Orders />)).toMatchSnapshot();
    }); 
});