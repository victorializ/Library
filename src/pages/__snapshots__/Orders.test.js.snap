// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders page component renders correctly 1`] = `
<Fragment>
  <Connect(User) />
  <Header>
    Your orders
  </Header>
  <div
    className="orders__list"
  >
    <LoadingComponent
      data={null}
      error={
        Object {
          "message": null,
        }
      }
      loading={true}
    >
      <OrdersList
        data={null}
      />
    </LoadingComponent>
  </div>
</Fragment>
`;
